Source/MonoGame.Extended/Timers/CountdownTimer.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.Xna.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace MonoGame.Extended.Timers
+{
+    public class CountdownTimer : IGameComponent
+    {
+        public event EventHandler Tick;
+        public event EventHandler Completed;
+
+        public TimerState TimerState;
+        public TimeSpan CurrentTick { get; set; } = TimeSpan.Zero;
+        public TimeSpan Interval { get; set; } = TimeSpan.Zero;
+        public TimeSpan TimeRemaining
+        {
+            get { return Interval - CurrentTick; }
+        }
+
+        public CountdownTimer(TimeSpan interval)
+        {
+            Interval = interval;
+            Reset();
+        }
+
+        void IGameComponent.Initialize() { Reset(); }
+        public void Reset()
+        {
+            TimerState = TimerState.Stopped;
+            CurrentTick = TimeSpan.Zero;
+        }
+
+        public virtual void Update(GameTime gameTime)
+        {
+            if (TimerState == TimerState.Stopped ||
+                TimerState == TimerState.Paused ||
+                TimerState == TimerState.Completed)
+                return;
+
+            CurrentTick += gameTime.ElapsedGameTime;
+
+            if (CurrentTick < Interval)
+            {
+                if (Tick != null)
+                    Tick(this, null);
```

Comment:
Probably best to get into a habit of firing events in a [thread safe way](http://stackoverflow.com/questions/3668953/raise-event-thread-safely-best-practice). It may never be needed, but it can't hurt.